src/utils/hooks/useProtectedPerson.ts

```diff
@@ -0,0 +1,18 @@
+import { useFetchProtectedPerson } from '../../api/hooks/useFetchProtectedPerson';
+import { CristinPerson } from '../../types/user.types';
+
+/**
+ * React hook that retrieves protected person data based on a CristinPerson object or person ID.
+ * Uses `useFetchProtectedPerson` to fetch data unless an existing person object is provided.
+ * @param existingPerson - CristinPerson object or person ID string.
+ * @param searchEnabled - Optional flag to enable or disable fetching (default: true).
+ * @returns An object containing the resolved person data and the query result.
+ */
+export const useProtectedPerson = (existingPerson: CristinPerson | string, searchEnabled = true) => {
```

Comment:
Min første tanke her er at dette fort blir en duplikat av `useFetchProtectedPerson`, så det kan være vanskelig å vite når man skal bruke hva. Om man sender inn personId (string) her er de vel i praksis helt lik, bortsett fra `searchEnabled = true` vs `{ enabled = true }` som argument. 

Personen man er ute etter vil også ligge i både `person` og `personQuery.data` bortsett fra når man sender inn en `person`, for da vil det ikke være behov for `useFetchProtectedPerson`.

Dermed var min tanke at det kanskje kunne vært en helper function som ev. bare gjør noe før man kaller `useFetchProtectedPerson` om det er behov. Men det er som sagt bare en tanke

Reply:
Ja, vi kan diskutere det med @iBuzza  og @joachimjorgensen . Tanken min var bare å wrappe all logikken rundt det at en existingPerson kan være streng (da må vi fetche) eller objekt (trenger ikke fetche). 

Altså bruker vi denne hooken er alt vi trenger å vite at vi sender inn existingPerson og om det er lov å gjøre oppslag(*) og får tilbake en person. Vi trenger ikke vite hva som eventuelt er gjort. Om vi gjorde oppslag eller ikke. Vi bare vet at vi har en person nå.

(*) Om det er lov å gjøre oppslag er om vi er inne i popup'en for eksempel.